test/src/server/routes/collection.js

```diff
@@ -160,3 +161,511 @@ describe('POST /collection/collectionID/delete', () => {
+		};
+		const response = await agent.post(`/collection/${collection.get('id')}/add`).send(data);
+		const item = await new UserCollectionItem().where('collection_id', collection.get('id')).fetchAll({require: false});
+		const itemJSON = item.toJSON();
+
+		expect(response.status).to.equal(403);
+		expect(response.res.statusMessage).to.equal('You do not have permission to edit this collection');
+		expect(itemJSON.length).to.equal(0);
+	});
+
+	it('should throw an error when trying to add entity with invalid bbid', async () => {
+		const collectionData = {
+			description: 'description',
+			entityType: 'Author',
+			name: 'collection name',
+			ownerId: loggedInUser.get('id'),
+			public: true
+		};
+		const collection = await new UserCollection(collectionData).save(null, {method: 'insert'});
+		const data = {
+			bbids: ['not-a-bbid']
+		};
+		const response = await agent.post(`/collection/${collection.get('id')}/add`).send(data);
+		const item = await new UserCollectionItem().where('collection_id', collection.get('id')).fetchAll({});
+		const itemJSON = item.toJSON();
+		expect(response).to.have.status(400);
+		expect(response.res.statusMessage).to.equal('Trying to add an entity having invalid BBID');
+		expect(itemJSON.length).to.equal(0);
+	});
+
+	it('should throw an error when trying to add an entity that does not exist', async () => {
+		const collectionData = {
+			description: 'description',
+			entityType: 'Author',
+			name: 'collection name',
+			ownerId: loggedInUser.get('id'),
+			public: true
+		};
+		const collection = await new UserCollection(collectionData).save(null, {method: 'insert'});
+		const data = {
+			bbids: [uuidv4()]
+		};
+		const response = await agent.post(`/collection/${collection.get('id')}/add`).send(data);
+		const item = await new UserCollectionItem().where('collection_id', collection.get('id')).fetchAll({});
+		const itemJSON = item.toJSON();
+		expect(response).to.have.status(400);
+		expect(response.res.statusMessage).to.equal('Trying to add an entity which is not present');
+		expect(itemJSON.length).to.equal(0);
+	});
+
+	it('should throw an error when trying to add wrong entity type', async () => {
+		const collectionData = {
+			description: 'description',
+			entityType: 'Edition',
+			name: 'collection name',
+			ownerId: loggedInUser.get('id'),
+			public: true
+		};
+		const collection = await new UserCollection(collectionData).save(null, {method: 'insert'});
+		const author = await createAuthor();
+		const data = {
+			bbids: [author.get('bbid')]
+		};
+		const response = await agent.post(`/collection/${collection.get('id')}/add`).send(data);
+		const item = await new UserCollectionItem().where('collection_id', collection.get('id')).fetchAll({});
+		const itemJSON = item.toJSON();
+
+		expect(response).to.have.status(400);
+		expect(response.res.statusMessage).to.equal('Trying to add an entity of type Author to a collection of type Edition');
+		expect(itemJSON.length).to.equal(0);
+	});
+});
```

Comment:
Another good test I hadn't thought about, but came up while looking at the middleware: what happens if I send an empty array? Same for the /remove endpoint.